File: Java-Spring/src/main/java/org/codecop/lcdnumbers/NumeralSystemProvider.java
package org.codecop.lcdnumbers;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class NumeralSystemProvider {

    @Value("${numeral_system_base}")
    private int base;

    @Bean
    public NumeralSystem createNumeralSystem() {
        return new NumeralSystemImpl(base);
    }
}

File: Kotlin-Spring/src/main/kotlin/org/codecop/lcdnumbers/NumeralSystemProvider.kt
package org.codecop.lcdnumbers

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class NumeralSystemProvider {
    @Value("\${numeral_system_base}")
    private val base = 0;

    @Bean
    fun createNumeralSystem(): NumeralSystem {
        return NumeralSystemImpl(base)
    }
}